### ISSUE TITLE
current stochastic gradient descent function trains slowly, and with great instability with higher learning rates

### ISSUE THREAD

**Issue by lucianojamimbo** [lucianojamimbo/ljmnist#1]

**Comment by lucianojamimbo** [comment]
fixed missing lines of code in the function trainSGD in commit ea0e6dabe7594a28c58a06be952a304db3ed6bc5

there may be more problems to solve to get SGD to work optimally so i will leave this issue open for now

**Comment by lucianojamimbo** [comment]
trainSGD now appears to behave as expected after removal of division of eta by batch size, and normalisation of input data.

74b7ba6af44fc2634983ffc27af16de6ccbea493

71c44da14efb13b96a2a24f61f341c64bf124fe6
